src/ThemeJsonExtractor.php

```diff
@@ -0,0 +1,183 @@
+							continue;
+						}
+
+						$translation = $translations->insert( $context, $array_to_translate[ $item_key ][ $key ] );
+						$translation->addReference( $file );
+					}
+				}
+			} else {
+				$array_to_translate = self::array_get( $theme_json, $path, null );
+				if ( is_null( $array_to_translate ) ) {
+					continue;
+				}
+
+				foreach ( $array_to_translate as $item_key => $item_to_translate ) {
+					if ( empty( $item_to_translate[ $key ] ) ) {
+						continue;
+					}
+
+					$translation = $translations->insert( $context, $array_to_translate[ $item_key ][ $key ] );
+					$translation->addReference( $file );
+				}
+			}
+		}
+	}
+
+	private static function read_json_file( $file_path ) {
+		$config = array();
+		if ( $file_path ) {
+			$decoded_file = json_decode(
+				file_get_contents( $file_path ),
+				true
+			);
+
+			$json_decoding_error = json_last_error();
+			if ( JSON_ERROR_NONE !== $json_decoding_error ) {
+				WP_CLI::debug( "Error when decoding {$file_path}", 'make-pot' );
+
+				return $config;
+			}
+
+			if ( is_array( $decoded_file ) ) {
+				$config = $decoded_file;
+			}
+		}
+		return $config;
+	}
+
+	private static function get_fields_to_translate() {
+		$file_structure  = self::read_json_file( __DIR__ . '/theme-i18n.json' );
+		$theme_json_i18n = self::extract_paths_to_translate( $file_structure );
+		return $theme_json_i18n;
+	}
+
+	private static function extract_paths_to_translate( $i18n_partial, $current_path = array() ) {
+		$result = array();
+		foreach ( $i18n_partial as $property => $partial_child ) {
+			if ( is_numeric( $property ) ) {
+				foreach ( $partial_child as $key => $context ) {
+					return array(
+						array(
+							'path'    => $current_path,
+							'key'     => $key,
+							'context' => $context,
+						),
+					);
+				}
+			}
+			$result = array_merge(
+				$result,
+				self::extract_paths_to_translate( $partial_child, array_merge( $current_path, array( $property ) ) )
+			);
+		}
+		return $result;
+	}
+
+	/**
+	 * Accesses an array in depth based on a path of keys.
+	 *
+	 * It is the PHP equivalent of JavaScript's `lodash.get()` and mirroring it may help other components
+	 * retain some symmetry between client and server implementations.
+	 *
+	 * Example usage:
+	 *
+	 *     $array = array(
+	 *         'a' => array(
+	 *             'b' => array(
+	 *                 'c' => 1,
+	 *             ),
+	 *         ),
+	 *     );
+	 *     array_get( $array, array( 'a', 'b', 'c' ) );
+	 *
+	 * @param array $array   An array from which we want to retrieve some information.
+	 * @param array $path    An array of keys describing the path with which to retrieve information.
+	 * @param mixed $default The return value if the path does not exist within the array,
+	 *                       or if `$array` or `$path` are not arrays.
+	 *
+	 * @return mixed The value from the path specified.
+	 */
+	private function array_get( $array, $path, $default = null ) {
```

Comment:
We might want to move those utils to https://github.com/wp-cli/wp-cli or at least extract them to a new abstract class so it can be used by `BlockExtractor`

Reply:
Happy to create a class for this to be available in other places. I don't see any potential use of them in `BlockExtractor`, though, so perhaps we can extract them when we need them?